src/CreatePoserFactory.php

```diff
@@ -48,9 +49,22 @@ protected function createFactory($factoryName, $className = null)
         $this->info("Creating Poser Factory called " . $factoryName);
 
         $expectedModelNameSpace = '\\' . modelsNamespace() . Str::beforeLast($factoryName, 'Factory');
-        $linkedModelNamespace = $this->option('model')
-            ? '\\' . $this->qualifyClass($this->option('model'))
-            : $expectedModelNameSpace;
+
+        // TODO - Decide if this actually should be caught
+        if (($optionModel = $this->option('model')) !== null) {
+            try {
+                $modelReflection = new \ReflectionClass($optionModel);
+                $linkedModelNamespace = '\\' . $modelReflection->getName();
+            } catch (ReflectionException $e) {
+                $linkedModelNamespace = $expectedModelNameSpace;
+            }
+        } else {
+            $linkedModelNamespace = $expectedModelNameSpace;
+        }
+
+        // TODO - or just choose this instead
```

Comment:
Much prefer this method. Cleaner, easier to understand, and I hate `if` statements 🤣